azurerm/internal/services/network/tests/security_partner_provider_resource_test.go

```diff
@@ -0,0 +1,229 @@
+		if rs.Type != "azurerm_security_partner_provider" {
+			continue
+		}
+
+		id, err := parse.SecurityPartnerProviderID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		if resp, err := client.Get(ctx, id.ResourceGroup, id.Name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("bad: Get on Network.SecurityPartnerProviderClient: %+v", err)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMSecurityPartnerProvider_template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-vhub-%d"
+  location = "%s"
+}
+
+resource "azurerm_virtual_wan" "test" {
+  name                = "acctest-vwan-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+}
+
+resource "azurerm_virtual_hub" "test" {
+  name                = "acctest-VHUB-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  virtual_wan_id      = azurerm_virtual_wan.test.id
+  address_prefix      = "10.0.2.0/24"
+}
+
+resource "azurerm_vpn_gateway" "test" {
+  name                = "acctest-VPNG-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  virtual_hub_id      = azurerm_virtual_hub.test.id
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger, data.RandomInteger)
+}
+
+func testAccAzureRMSecurityPartnerProvider_basic(data acceptance.TestData) string {
+	template := testAccAzureRMSecurityPartnerProvider_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_security_partner_provider" "test" {
+  name                   = "acctest-spp-%d"
+  resource_group_name    = azurerm_resource_group.test.name
+  location               = azurerm_resource_group.test.location
+  security_provider_type = "ZScaler"
+
+  depends_on = [azurerm_vpn_gateway.test]
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMSecurityPartnerProvider_requiresImport(data acceptance.TestData) string {
+	config := testAccAzureRMSecurityPartnerProvider_basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_security_partner_provider" "import" {
+  name                   = azurerm_security_partner_provider.test.name
+  resource_group_name    = azurerm_security_partner_provider.test.resource_group_name
+  location               = azurerm_security_partner_provider.test.location
+  security_provider_type = azurerm_security_partner_provider.test.security_provider_type
+}
+`, config)
+}
+
+func testAccAzureRMSecurityPartnerProvider_complete(data acceptance.TestData) string {
+	template := testAccAzureRMSecurityPartnerProvider_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_security_partner_provider" "test" {
+  name                   = "acctest-spp-%d"
```

Comment:
Could we ensure capitals work
```suggestion
  name                   = "acctest-SPP-%d"
```

Reply:
updated